File: Batalla_Poo_Ruby/soldado.rb
require_relative "participante.rb"

class Soldado < Participante

    def getTipo
        return "Soldado"
    end

    def disparar(enemigo)
       enemigo.danio(enemigo.danioParaSoldado())
    end

    def danioParaTanque()
        return 700
    end

    def danioParaSoldado()
        return 200
    end

    def danioParaAvion()
        return 450
    end

end

File: Batalla_Poo_CPP/src/Soldado.cpp
#include "Soldado.h"


Soldado::Soldado():Participante()
{
    this->setNombre("Soldado");
}

/*
void Soldado::disparar(Participante enemigo)
{
    enemigo.quitarVida(enemigo.danioParaSoldado());
}
*/

int Soldado::danioParaTanque()
{
    return 700;
}

int Soldado::danioParaAvion()
{
    return 200;
}

int Soldado::danioParaSoldado()
{
    return 450;
}
